scrapers/twitter.js

```diff
@@ -1,34 +1,115 @@
-		this.request();
+		this.REQUEST_URL = this.BASE_URL + this.SLICED_PROC_ARGS + "&src=typd";
 	}
 
 	scrape($) {
 		// Scraping logic goes here.
 		// We have our HTML and $ set here so we can directly use these.
 		// Set the result as this.JSON = {}
+		const tweets = [];
+		$("div.tweet").each( (i, element) => {
+			let tweetDetail = {};
+
+			// userid, avatarurl, fullname, nickname
+			tweetDetail.userid = $(element).attr("data-user-id");
+			tweetDetail.useravatarurl = $(element).find("img.avatar").attr("src");
+			tweetDetail.userfullname = $(element).attr("data-name");
+			tweetDetail.usernickname = $(element).attr("data-screen-name");
+
+			// status_url, time-name, time_in_millis
+			tweetDetail.tweetstatusurl = $(element).find("a.tweet-timestamp").attr("href");
+			tweetDetail.tweettimename = $(element).find("a.tweet-timestamp").attr("title");
+			tweetDetail.tweettimems = $(element).find("span._timestamp").attr("data-time-ms");
+
+			// tweet_text
+			tweetDetail.tweettext = $(element).find("p.tweet-text").text();
+
+			// TODO: parsing_retweet, parsing_favourite
+			$(element).find("span.ProfileTweet-actionCountForPresentation").each((i, ele) => {
+				let number = $(ele).text();
+				if (i === 1) {
+					if (number.length > 0) {
+						tweetDetail.tweetretweetcount = parseInt(number);
+					} else {
+						tweetDetail.tweetretweetcount = 0;
+					}
+				}
+				if (i === 3) {
+					if (number.length > 0) {
+						tweetDetail.tweetfavouritecount = parseInt(number);
+					} else {
+						tweetDetail.tweetfavouritecount = 0;
+					}
+				}
+			});
+
+			// tweet images
+			tweetDetail.images = []
+			$(element).find("img").each((i, ele) => {
+				let imageLink = $(ele).attr("src");
+				tweetDetail.images.push(imageLink);
+			});
+			tweetDetail.images.splice(0, 1); // removes the profile image url
+
+			// TODO: add location data
+			// TODO: video, audio, gifs
+			tweets.push(tweetDetail);
+		});
+		
+		tweets.splice(-1, 1); // all data is not available for the last tweet
+		
+		this.JSON = {};
+		this.JSON.statuses = tweets;
+		return tweets;
+	}
+
+	/**
+	 * Uses the scrape method to scrape Tweets from twitter and passes the data
+	 * to callback function.
+	 * @param {*string} query Tweet search query
+	 * @param {*function} callback callback function to be invoked after completion
+	 */
+	getTweets(query, callback) {
+		let url = this.BASE_URL + query + "&src=typd";
+		let options = { uri: url};
+
+		request(options)
+			.then((body) => {
+				let html = body;
+				let $ = cheerio.load(html);
+
+				let tweets = this.scrape($);
+				callback(tweets);
+			})
+			.catch((error) => {
+				callback(error.message);
+			})
 	}
 }
 
 module.exports = TwitterLoklakScrapper;
 
-new TwitterLoklakScrapper();
+// uncomment the following to test
+// let query = process.argv[2];
+// let scraper = new TwitterLoklakScrapper();
+// scraper.getTweets(query, (data) => {
+// 	console.log(data);
+// 	console.log(data.length);
+// });
```

Comment:
Please insert a newline in the end. Always configure your IDE for `.editorconfig` to make things smooth.